pkg/pull/pull.go

```diff
@@ -440,9 +440,12 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		chartBaseName := v.Spec.Chart.Name
 		// the helmBase may have a chart name prefix removed - we must find the base name instead of the original chart name
 		for _, helmBase := range helmBases {
-			chartName := strings.Split(helmBase.Path, "/")[len(strings.Split(helmBase.Path, "/"))-1]
-			if strings.HasSuffix(v.Spec.Chart.Name, chartName) {
+			baseDir := filepath.Dir(helmBase.Path)
+			chartName := filepath.Base(helmBase.Path)
```

Comment:
i think the base path comes from `metadata.name` or the new `releaseName` field, not the chart name. so comparing this to `v.Spec.Chart.Name` is not accurate i believe

Reply:
there's a `v.GetDirName()` function that should return the base directory name that was used